core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-favorite-page/dot-favorite-page.component.ts

```diff
@@ -0,0 +1,123 @@
+import { Observable, Subject } from 'rxjs';
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { FormGroup, Validators, UntypedFormBuilder } from '@angular/forms';
+import { DynamicDialogConfig, DynamicDialogRef } from 'primeng/dynamicdialog';
+import { map, startWith, takeUntil } from 'rxjs/operators';
+import { DotRole } from '@dotcms/app/shared/models/dot-role/dot-role.model';
+import { DotPageRenderState } from '../../shared/models';
+import { DotFavoritePageState, DotFavoritePageStore } from './store/dot-favorite-page.store';
+
+export interface DotFavoritePageProps {
+    pageRenderedHtml?: string;
+    pageState: DotPageRenderState;
+}
+
+export interface DotFavoritePageFormData {
+    currentUserRoleId: string;
+    thumbnail?: string;
+    title: string;
+    url: string;
+    order: number;
+    permissions?: DotRole[];
+}
+
+@Component({
+    selector: 'dot-favorite-page',
+    templateUrl: 'dot-favorite-page.component.html',
+    providers: [DotFavoritePageStore]
+})
+export class DotFavoritePageComponent implements OnInit, OnDestroy {
+    form: FormGroup;
+    isFormValid$: Observable<boolean>;
+    pageRenderedHtml: string;
+
+    vm$: Observable<DotFavoritePageState> = this.store.vm$;
+
+    private destroy$: Subject<boolean> = new Subject<boolean>();
+
+    constructor(
+        private ref: DynamicDialogRef,
+        private config: DynamicDialogConfig,
+        private fb: UntypedFormBuilder,
+        private store: DotFavoritePageStore
+    ) {
+        this.store.setInitialStateData(this.config.data.page);
+        this.pageRenderedHtml = this.config.data.page.pageRenderedHtml;
+    }
+
+    ngOnInit(): void {
+        const { page }: { page: DotFavoritePageProps } = this.config.data;
+
+        this.form = this.fb.group(this.setupForm(page));
+        this.isFormValid$ = this.form.valueChanges.pipe(
+            takeUntil(this.destroy$),
+            map(() => {
+                return this.form.valid;
+            }),
+            startWith(false)
+        );
+
+        this.store.currentUserRoleId$
+            .pipe(takeUntil(this.destroy$))
+            .subscribe((currentUserRoleId: string) => {
+                this.form.get('currentUserRoleId').setValue(currentUserRoleId);
+            });
+
+        this.store.closeDialog$.pipe(takeUntil(this.destroy$)).subscribe((closeDialog: boolean) => {
+            if (closeDialog) {
+                this.ref.close(true);
+            }
+        });
+
+        // This is needed to wait until the Web Component is rendered
+        setTimeout(() => {
```

Comment:
Maybe we can use `requestAnimationFrame` instead of `setTimeout`. [doc](https://developer.mozilla.org/es/docs/Web/API/Window/requestAnimationFrame)